drivers/sensors/cirque_pinnacle.c

```diff
@@ -200,39 +200,84 @@ void cirque_pinnacle_init(void) {
 #endif
 
     touchpad_init = true;
+
     // Host clears SW_CC flag
     cirque_pinnacle_clear_flags();
 
-    // Host configures bits of registers 0x03 and 0x05
-    RAP_Write(SYSCONFIG_1, SYSCONFIG_1_VALUE);
-    RAP_Write(FEEDCONFIG_2, FEEDCONFIG_2_VALUE);
-
-    // Host enables preferred output mode (absolute)
-    RAP_Write(FEEDCONFIG_1, FEEDCONFIG_1_VALUE);
+    // SysConfig1 (Low Power Mode)
+    // Bit 0: Reset, 1=Reset
+    // Bit 1: Shutdown, 1=Shutdown, 0=Active
+    // Bit 2: Sleep Enable, 1=low power mode, 0=normal mode
+    // send a RESET command now, in case QMK had a soft-reset without a power cycle
+    RAP_Write(SYSCONFIG_1, 0x01);
+    wait_ms(20); // Pinnacle needs 10-15ms to boot, so wait long enough before configuring
+    RAP_Write(SYSCONFIG_1, 0x00);
+    wait_us(50);
 
-    // Host sets z-idle packet count to 5 (default is 30)
-    RAP_Write(Z_IDLE_COUNT, Z_IDLE_COUNT_VALUE);
+    // FeedConfig2 (Feature flags for Relative Mode Only)
+    // Bit 0: IntelliMouse Enable, 1=enable, 0=disable
+    // Bit 1: All Taps Disable, 1=disable, 0=enable
+    // Bit 2: Secondary Tap Disable, 1=disable, 0=enable
+    // Bit 3: Scroll Disable, 1=disable, 0=enable
+    // Bit 4: GlideExtend® Disable, 1=disable, 0=enable
+    // Bit 5: reserved
+    // Bit 6: reserved
+    // Bit 7: Swap X & Y, 1=90° rotation, 0=0° rotation
+    RAP_Write(FEEDCONFIG_2, 0x00);
+
+    // FeedConfig1 (Data Output Flags)
+    // Bit 0: Feed enable, 1=feed, 0=no feed
+    // Bit 1: Data mode, 1=absolute, 0=relative
+    // Bit 2: Filter disable, 1=no filter, 0=filter
+    // Bit 3: X disable, 1=no X data, 0=X data
+    // Bit 4: Y disable, 1=no Y data, 0=Y data
+    // Bit 5: reserved
+    // Bit 6: X data Invert, 1=X max to 0, 0=0 to Y max
+    // Bit 7: Y data Invert, 1=Y max to 0, 0=0 to Y max
+    RAP_Write(FEEDCONFIG_1, CIRQUE_PINNACLE_POSITION_MODE << 1);
+
+    // Host sets z-idle packet count to 5 (default is 0x1F/30)
+    RAP_Write(Z_IDLE_COUNT, 0);
 
     cirque_pinnacle_set_adc_attenuation(0xFF);
     cirque_pinnacle_tune_edge_sensitivity();
     cirque_pinnacle_enable_feed(true);
 }
 
-// Reads XYZ data from Pinnacle registers 0x14 through 0x17
-// Stores result in pinnacle_data_t struct with xValue, yValue, and zValue members
 pinnacle_data_t cirque_pinnacle_read_data(void) {
-    uint8_t         data[6] = {0};
-    pinnacle_data_t result  = {0};
+    uint8_t         data_ready = 0;
+    uint8_t         data[6]    = {0};
+    pinnacle_data_t result     = {0};
+
+    // Check if there is valid data available
+    RAP_ReadBytes(STATUS_1, &data_ready, 1); // bit2 is Software Data Ready, bit3 is Command Complete, bit0 and bit1 are reserved/unused
+    if ((data_ready & 0x04) == 0) {
+        // no data available yet
+        return result;
```

Comment:
How does this affect the caller? If this function always checks for data readiness, maybe it should have a return code to explicitly indicate data not ready. Otherwise, I'd prefer to have a separate data ready check function and let the caller decide whether to use it or not.

Reply:
yes, that makes sense - I'll introduce a new attribute in `pinnacle_data_t`: `bool valid` to store this information